platform/ios/src/MGLScaleControlView.m

```diff
@@ -3,13 +3,25 @@
 @interface MGLScaleControlView()
 @property (nonatomic, assign) double zoomLevel;
 @property (nonatomic, assign) CLLocationDistance metersPerPoint;
-@property (nonatomic) NSArray *labels;
-@property (nonatomic) NSArray *bars;
+@property (nonatomic) NSArray<UILabel *> *labels;
+@property (nonatomic) NSArray<UIView *> *bars;
 @property (nonatomic) UILabel *debugLabel;
+@property (nonatomic) NSLengthFormatter *formatter;
 @end
 
 @implementation MGLScaleControlView
 
+static const NSUInteger TEMP_NUMBER_OF_BARS = 2;
+static const CGFloat BASE_WIDTH = 100;
+
+- (instancetype)init {
+    if (self == [super init]) {
+        _formatter = [[NSLengthFormatter alloc] init];
```

Comment:
We should use MGLDistanceFormatter once #3331 is implemented. NSLengthFormatter is suboptimal because, when the system is set to U.S. units, it tends to use yards instead of feet or fractional miles, whereas Americans find yards difficult to relate to (except in football).